### DISCUSSION TITLE
Cookie is parsed as empty object

### DISCUSSION THREAD

**Discussion by tjhoo** [remix-run/remix#10048] [Q&A] [answered]
Hi

I'm trying to read a cookie set by another web application,
```javascript

export const authCookie = createCookie("auth", {
  maxAge: 60 * 60 * 1000,
});

export const loader = async ({ request }) => {
  const header = await request.headers.get("Cookie");
  console.log(header)   // print auth=hello
  const auth = await authCookie.parse(header);
  console.log(auth)   // print {}
  ...
};
```
The first console log prints `auth=hello` and this shows that the cookie header is available. However, it returned `{}` when parsed. May I know what am I missing here?

**Comment by kiliman** [comment, marked as answer]
The Remix cookie functions are designed to encode/decode cookies created by Remix, not generic cookies. These cookies are typically base-64 encoded and signed, so if not in the expected format, Remix will return an empty object.

If you want to support 3rd party cookies, then you'll need to parse them yourself. You can use the `cookie` package, which is the same library that Remix uses.

Here's more discussion about it.

https://github.com/remix-run/remix/discussions/5198
